core/resolver/uri/uri.go

```diff
@@ -1,7 +1,42 @@
 package uri
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
 type URI struct {
 	Authority string
 	Path      string
 	Uri       string
 }
+
+var (
+	authorities = map[string]bool{"ipfs": true, "ens": true, "fs": true, "uns": true}
+	re          = regexp.MustCompile("^(wrap://)*(?P<authority>[a-z][a-z0-9-_]+)/(?P<path>.*)$")
+)
+
+func New(uri string) (*URI, error) {
+	exp := regexp.MustCompile("^/*")
+	processed := exp.Split(uri, 1)[0]
```

Comment:
in python and rust version used function "lstrip", in golang it is "strings.TrimLeft".